### PR TITLE
[KED-2314] Support launching a development server against a real pipeline

### PR DESCRIPTION
**Pull request by limdauto**
## Description

Support launching a development server against a real Kedro pipeline by adding a `start:api` command to npm. Feel free to add it to the main `npm start` if you'd like :) 

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "start:app": "PORT=4141 react-scripts start",
```
Comment (studioswong) [inline package.json:14]: Thanks for setting this up @limdauto - I notice that your changes mentioned running viz from port 4142 while the proxy changes haven't been applied here yet - I assume you intend for this PR to be merged in after @bru5 's other PR?

Comment (bru5) [reply package.json:14]: Yeah I'd agree I think add the proxy in this PR too as this appears to rely on that? Should be fine from a merge perspective.

Comment (limdauto) [reply package.json:14]: Yea proxy is already on `main`. Can either of you approve this PR please?

**Review by bru5 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (studioswong) [reply package.json:14]: sounds great, thanks both!

**Pull request merged**